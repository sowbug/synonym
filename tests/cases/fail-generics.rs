use synonym::Synonym;

#[derive(Synonym)]
pub struct Foo<T> {
    _foo: std::marker::PhantomData<T>
}

fn main() {}
