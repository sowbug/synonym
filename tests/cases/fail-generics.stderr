error: proc-macro derive panicked
 --> $DIR/fail-generics.rs:3:10
  |
3 | #[derive(Synonym)]
  |          ^^^^^^^
  |
  = help: message: Synonym supports only tuple structs with single unnamed field. Enums, unions and generic type parameters are not supported
